app/Models/User.php

```diff
@@ -1814,6 +1814,39 @@ public function titleUrl(): ?string
         return $this->rank?->url;
     }
 
+    public function toMetaDescription(array $options = []): string
+    {
+        static $rankTypes = ['country', 'global'];
+
+        $ruleset = $options['ruleset'] ?? $this->playmode;
+        $stats = $this->statistics($ruleset);
+
+        $replacements['ruleset'] = $ruleset;
+
+        foreach ($rankTypes as $type) {
+            $method = "{$type}Rank";
+            $replacements[$type] = osu_trans("users.ogp.description.{$type}", [
+                'rank' => format_rank($stats?->$method()),
+            ]);
+
+            $variants = Beatmap::VARIANTS[$ruleset] ?? [];
+
+            $variantsTexts = null;
+            foreach ($variants as $variant) {
+                $variantStats = $this->statistics($ruleset, false, $variant);
+                if ($variantStats !== null) {
+                    $variantsTexts[] = $variant.' '.format_rank($variantStats->$method());
+                }
```

Comment:
I'll need to check again but there may be case where the variant entry exists but not ranked.